server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/webhook/WebhookPostTask.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.computation.task.projectanalysis.webhook;
+
+import java.util.Optional;
+import org.sonar.api.ce.posttask.PostProjectAnalysisTask;
+import org.sonar.api.config.Settings;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonar.core.config.WebhookProperties;
+import org.sonar.server.computation.task.projectanalysis.component.SettingsRepository;
+import org.sonar.server.computation.task.projectanalysis.component.TreeRootHolder;
+
+import static com.google.common.base.Throwables.getRootCause;
+import static java.lang.String.format;
+
+public class WebhookPostTask implements PostProjectAnalysisTask {
+
+  private static final Logger LOGGER = Loggers.get(WebhookPostTask.class);
+
+  private final TreeRootHolder rootHolder;
+  private final SettingsRepository settingsRepository;
+  private final WebhookSender sender;
+
+  public WebhookPostTask(TreeRootHolder rootHolder, SettingsRepository settingsRepository, WebhookSender sender) {
+    this.rootHolder = rootHolder;
+    this.settingsRepository = settingsRepository;
+    this.sender = sender;
+  }
+
+  @Override
+  public void finished(ProjectAnalysis analysis) {
+    process(analysis, WebhookProperties.GLOBAL_KEY_PREFIX);
+    process(analysis, WebhookProperties.PROJECT_KEY_PREFIX);
+  }
+
+  private void process(ProjectAnalysis analysis, String propertyPrefix) {
+    Settings settings = settingsRepository.getSettings(rootHolder.getRoot());
```

Comment:
I don't agree with "this is not all part of the contract of the interface". It should be the opposite. It should be documented if method can't be called multiple times.

In my case, I don't see any performance issue here, so I prefer improving readability of code (avoid passing settings as a parameter).